Mathlib/Topology/Path/CountableConcat.lean

```diff
@@ -0,0 +1,228 @@
+        · refine Iic_mem_nhds <| Subtype.coe_lt_coe.1 (?_ : t.1 < 2⁻¹)
+          rw [Nat.log_eq_zero_iff, or_iff_left one_lt_two.not_le, Nat.floor_lt (inv_pos.2 ht'').le,
+            inv_lt_comm₀ (by exact ht'') two_pos, coe_symm_eq, lt_sub_comm] at h
+          exact h.trans_eq (by ring)
+      | succ n =>
+        refine ContinuousOn.continuousAt (s := Set.Icc
+          ⟨1 - (2 ^ n)⁻¹, by simp [inv_le_one_of_one_le₀ <| one_le_pow₀ one_le_two (M₀ := ℝ)]⟩
+          ⟨1 - (2 ^ (n + 2))⁻¹, by
+            simp [inv_le_one_of_one_le₀ <| one_le_pow₀ one_le_two (M₀ := ℝ)]⟩) ?_ ?_
+        · convert (hγ' n).union_isClosed isClosed_Icc isClosed_Icc <| hγ' (n + 1) using 1
+          rw [add_assoc, one_add_one_eq_two, Set.Icc_union_Icc_eq_Icc]
+          · rfl
+          · simp only [symm_le_symm, Subtype.mk_le_mk]
+            exact inv_anti₀ (by simp) <| pow_le_pow_right₀ one_le_two (by simp)
+          · simp only [symm_le_symm, Subtype.mk_le_mk]
+            exact inv_anti₀ (by simp) <| pow_le_pow_right₀ one_le_two (by simp)
+        · refine Icc_mem_nhds ?_ ?_ <;> rw [← Subtype.coe_lt_coe, Subtype.coe_mk]
+          · replace h := h.symm.le; rw [← Nat.pow_le_iff_le_log one_lt_two (Nat.floor_pos.2 <|
+              (one_le_inv₀ ht'').2 (σ t).2.2).ne', Nat.le_floor_iff (inv_pos.2 ht'').le,
+              le_inv_comm₀ (by simp) ht'', coe_symm_eq, sub_le_comm] at h
+            refine (sub_lt_sub_left (inv_strictAnti₀ (by simp) ?_) 1).trans_le h
+            rw [Nat.cast_pow, Nat.cast_ofNat]
+            exact pow_lt_pow_right₀ one_lt_two n.lt_succ_self
+          · replace h := h.trans_lt (Nat.lt_succ_self _); rw [← Nat.lt_pow_iff_log_lt one_lt_two
+              (Nat.floor_pos.2 <| (one_le_inv₀ ht'').2 (σ t).2.2).ne', Nat.floor_lt
+              (inv_pos.2 ht'').le, inv_lt_comm₀ ht'' (by simp), coe_symm_eq, lt_sub_comm] at h
+            exact h.trans_eq <| by simp
+    · rw [unitInterval.lt_one_iff_ne_one, not_ne_iff] at ht; rw [ht]
+      unfold ContinuousAt
+      intro u hu
+      rw [countableConcatFun_one] at hu
+      let ⟨n, hn⟩ : ∃ n : ℕ, ∀ m, n ≤ m → ∀ t, γ m t ∈ u := by
+        simpa [tendsto_def, mem_prod_top] using hγx hu
+      rw [mem_map, mem_nhds_iff]
+      use Set.Ioi ⟨1 - (2 ^ n)⁻¹, by rw [sub_nonneg, inv_le_one₀] <;> simp [one_le_pow₀], by simp⟩
+      refine ⟨fun t ht ↦ ?_, isOpen_Ioi, by simp [← coe_lt_one]⟩
+      by_cases ht' : t < 1
+      · have ht'' := symm_one ▸ symm_lt_symm.2 ht'; have ht''' := coe_pos.2 ht''
+        simp only [mem_preimage, countableConcatFun, ht', reduceDIte]
+        refine hn _ ?_ _
+        rw [← Nat.pow_le_iff_le_log one_lt_two (Nat.floor_pos.2 <| (one_le_inv₀ ht''').2
+          (σ t).2.2).ne', Nat.le_floor_iff (inv_pos.2 ht''').le, le_inv_comm₀ (by simp) ht''',
+          coe_symm_eq, sub_le_comm]
+        apply le_of_lt; simpa using ht
+      · rw [unitInterval.lt_one_iff_ne_one, not_ne_iff] at ht'; rw [ht']
+        rw [mem_preimage, countableConcatFun_one]; exact mem_of_mem_nhds hu
```

Comment:
Here's the trick now: what you are doing here is basically reproving that limits compose. This means you should be able to golf this by proving that the map you compose `γ` with, let's call it `f : I → ℕ × I`, tends to `atTop ×ˢ ⊤` along `𝓝 (1 : I)`, ie you should...